src/main/java/frc/robot/Autonomous.java

```diff
@@ -41,42 +36,54 @@
 
     private final PIDController positionPIDController;
 
+    private final CANSparkMax leftSide;
+    private final CANSparkMax rightSide;
+
     public Autonomous(Waiter waiter, Drive drivetrain, BallPath ballPath){
         this.waiter = waiter;
         this.drivetrain = drivetrain;
+        this.leftSide = drivetrain.getLeftSide();
+        this.rightSide = drivetrain.getRightSide();
         this.ballPath = ballPath;
         this.positionPIDController = new PIDController(kP, kI, kD);
         this.positionPIDController.setTolerance(DRIVE_DIST_TOLERANCE);
     }
 
-    double calcTicks(double encoderTicks){
-        // Calculates ticks per revolution(shaft rotation)
-        double gearRatio = 8;
-
-        double ticks = encoderTicks / gearRatio * wheelCircumference; // ticksPer cancels so it's just encoderTicks / gearRatio
+    public void turn(AHRS gyro, double degree) {
+        gyro.reset();
+        double error = degree - gyro.getAngle();
+        double kP = 0.005;
+        double tolerance = 2;
+
+        while (gyro.getAngle() < degree - tolerance || gyro.getAngle() > degree + tolerance){
```

Comment:
If this is going to be a `while` loop on its own, since it's called from within the one-shot/top-down autonomous routine thread, then it would be good to insert a `Thread.sleep()` or `Timer.delay()` within the loop body. Otherwise, the CPU core will be spinning on this loop extremely rapidly and wasting power and compute time for no real benefit.

Reply:
20 milliseconds enough?

Reply:
Yea, should be good.